### PR TITLE
Log durations of different phases during process execution

### PR DESCRIPTION
**Pull request by almiliMSFT**
Phases for which duration is measured:
  - processing SandboxProcessResult
  - processing standard outputs
  - getting observed file accesses
  - logging standard outputs
  - reporting SandboxExecutionResult
  - validating observed file accesses
  - computing isDirty
  - storing cache contents
  - storing fingerprint computation to XLG
  - flagging shared opaque outputs
  - analyzing access violations
  - analyzing double writes
  - reporting output content

### CODE REVIEW

File: Public/Src/Engine/Processes/SandboxedProcessPipExecutor.cs [hunk 1]
```diff
@@ -1589,6 +1609,9 @@ private void LogCommandLineTooLong(SandboxedProcessInfo info)
             // If a PipProcessError was logged, the pip cannot be marked as succeeded
             Contract.Assert(!standardOutHasBeenWrittenToLog || status != SandboxedProcessPipExecutionStatus.Succeeded);
 
+            Tracing.Logger.Log.LogDetoursDebugMessage(m_loggingContext, m_pip.SemiStableHash, m_pip.GetDescription(m_context),
+                $"Done with the rest of the crap in {DateTime.UtcNow.Subtract(start)}");
```
Comment (Kr1sso) [inline Public/Src/Engine/Processes/SandboxedProcessPipExecutor.cs]: Nit: other word maybe? 💃

Comment (almiliMSFT) [reply Public/Src/Engine/Processes/SandboxedProcessPipExecutor.cs]: as I said, this wasn't meant to make it to a PR :)

Comment (almiliMSFT) [inline Public/Src/Engine/Processes/SandboxedProcessPipExecutor.cs]:
```suggestion
                $"Done with the rest of the stuff in {DateTime.UtcNow.Subtract(start)}");
```

**Comment by narasamdya** [comment]
Is this going to be checked in to the master?

File: Public/Src/Engine/Scheduler/Tracing/Log.cs [hunk 2]
```diff
@@ -1,6 +1,7 @@
 // Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+using System;
```
Comment (almiliMSFT) [inline Public/Src/Engine/Scheduler/Tracing/Log.cs]:
```suggestion
```

**Pull request merged**